packages/components/grid/src/grid.ts

```diff
@@ -326,34 +344,42 @@ export class Grid<T = any> extends ScopedElementsMixin(LitElement) {
         ${msg('Skip to end of table')}</a
       >
       <table part="table" aria-rowcount=${this.dataSource?.items.length || 0}>
+        <caption></caption>
         <thead
           @sl-filter-change=${this.#onFilterChange}
-          @sl-filter-value-change=${this.#onFilterValueChange}
-          @sl-sort-direction-change=${this.#onSortDirectionChange}
+          @sl-filter-register=${this.#onFilterRegister}
           @sl-sorter-change=${this.#onSorterChange}
+          @sl-sorter-register=${this.#onSorterRegister}
           part="thead"
         >
-          ${this.renderHeader()}
+          ${this.renderHeaderRows()}
         </thead>
         <tbody id="tbody" part="tbody">
           ${virtualize({
             items: this.view.rows,
             renderItem: (item, index) => this.renderItem(item, index)
           })}
         </tbody>
-        <tfoot>
-          ${this.scrollbar
-            ? html`
+        ${this.scrollbar
+          ? html`
+              <tfoot>
                 <tr class="scrollbar">
                   <td>
                     <sl-scrollbar scroller="tbody"></sl-scrollbar>
                   </td>
                 </tr>
-              `
-            : nothing}
-        </tfoot>
+              </tfoot>
+            `
+          : nothing}
       </table>
 
+      <div part="bulk-actions" popover="manual">
```

Comment:
When I selected few rows I couldn't focus anything inside bulk-actions, I couldn't get there with keyboard to focus anything inside.

![image](https://github.com/user-attachments/assets/7dbedac4-359a-4586-bd82-6478e774a74b)

Reply:
So the `focus-visible` is broken in the SL theme. If you switch to the Magister theme it works. Once you've tabbed through all the checkboxes, focus will go to the buttons in the popover.